pandas/_libs/tslib.pyx

```diff
@@ -580,58 +582,65 @@ cpdef array_to_datetime(ndarray[object] values, str errors='raise',
                         continue
 
                     try:
-                        _string_to_dts(val, &dts, &out_local, &out_tzoffset)
-                    except ValueError:
-                        # A ValueError at this point is a _parsing_ error
-                        # specifically _not_ OutOfBoundsDatetime
-                        if _parse_today_now(val, &iresult[i]):
-                            continue
-                        elif require_iso8601:
-                            # if requiring iso8601 strings, skip trying
-                            # other formats
-                            if is_coerce:
-                                iresult[i] = NPY_NAT
+                        string_to_dts_failed = _string_to_dts_noexc(
+                            val, &dts, &out_local,
+                            &out_tzoffset
```

Comment:
i think cython now supports specifying typed tuples for return types, e.g. `cdef (bint, int, int) _string_to_dts_no_exc(...)`.  That would be more idiomatic than pointer-passing.  Thoughts @WillAyd ?